ui/lib/sync/addon/components/secrets/page/destinations/destination/sync.hbs

```diff
@@ -0,0 +1,84 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+<SyncHeader
+  @title="Sync secrets to {{@destination.typeDisplayName}}"
+  @breadcrumbs={{array
+    (hash label="Secrets sync" route="secrets.overview")
+    (hash label="Destinations" route="secrets.destinations")
+    (hash label="Destination" route="secrets.destinations.destination")
+    (hash label="Sync")
+  }}
+/>
+
+<form {{on "submit" (perform this.setAssociation)}} class="has-top-margin-xl">
+  <p class="is-label">Which secrets would you like us to sync?</p>
+  <p class="sub-text">
+    Select a KV engine mount and path to sync a secret to the destination.
+  </p>
+
+  <MessageError @errorMessage={{this.error}} />
+
+  <div class="has-top-margin-l">
+    {{#if this.mounts}}
+      <SearchSelect
+        @label="Select a mount for the KV engine"
+        @options={{this.mounts}}
+        @selectLimit={{1}}
+        @disallowNewItems={{true}}
+        @onChange={{this.setMount}}
+        data-test-sync-mount-select
+      />
+    {{else}}
+      <InputSearch
+        @label="Enter an existing KV engine mount"
+        @inputValue={{this.mountPath}}
+        @onChange={{fn (mut this.mountPath)}}
+      />
+    {{/if}}
+
+    <FormFieldLabel
+      @label="Select a secret to sync"
+      @subText="Enter the full path to the secret. Suggestions will display below if permitted by policy."
+    />
+    <FilterInput
+      placeholder="Path to secret"
+      value={{this.secretPath}}
+      disabled={{not this.mountPath}}
+      @onInput={{this.onSecretInput}}
+      {{on "click" this.onSecretInputClick}}
+      data-test-sync-secret-input
+    />
+    <PowerSelect
+      @eventType="click"
+      @options={{this.secrets}}
+      @onChange={{this.onSuggestionSelect}}
+      @renderInPlace={{true}}
+      @closeOnSelect={{false}}
+      @disabled={{not this.mountPath}}
+      @registerAPI={{fn (mut this.powerSelectAPI)}}
+      @triggerClass="is-hidden"
+      @noMatchesMessage="No suggestions for this path"
+      data-test-sync-secret-select
+      as |secret|
+    >
+      {{secret.path}}
+    </PowerSelect>
```

Comment:
The use case here is outside of the scope of the `SearchSelect` component. When a selection is made, the input value is appended with the selected value and a query is made to fetch new results at the updated secret path. These new results are then rendered in the `PowerSelect` which remains open after a selection is made. We hope this improves the experience of selecting a secret at a nested path.

Reply:
We actually have a use case for this exact behavior in Dashboard, so this is an excellent improvement! And I agree, outside the scope of SearchSelect

Reply:
That's good to know! I was considering making it a component to begin with but wasn't sure if there were other use cases. I can follow up and move this out into a `SuggestionInput` component or something along those lines in a separate PR.